# APM111 - Statistical Theory - FA 6

In [27]:
import pandas as pd
import numpy as np

### 1. Table 1 shows a frequency distribution of grades on a final examination in college algebra. Find the quartiles of the distribution.

In [28]:
df = pd.read_csv('algebra.csv')
df

Grade  Number of Students
0  90-100                   9
1   80-89                  32
2   70-79                  43
3   60-69                  21
4   50-59                  11
5   40-49                   3
6   30-39                   1

In [29]:
def expand_grades(row):
    range_start, range_end = map(int, row['Grade'].split('-'))
    return [value for value in range(range_start, range_end + 1) for _ in range(row['Number of Students'])]

expanded_grades = df.apply(expand_grades, axis=1).explode().astype(int)

#### Quartiles

In [30]:
# Calculate quartiles
Q1 = expanded_grades.quantile(0.25)
Q2 = expanded_grades.quantile(0.5)
Q3 = expanded_grades.quantile(0.75)

print(f'''Quartiles 
25th Percentile: {Q1}
50th Percentile: {Q2}
75th Percentile: {Q3}''')

Quartiles 
25th Percentile: 67.0
50th Percentile: 75.0
75th Percentile: 83.0


### 2a. On a final examination in statistics, the mean grade of a group of 150 students was 78 and the standard deviation was 8.0. In algebra, however, the mean final grade of the group was 73 and the standard deviation was 7.6. In which subject was there the greater absolute dispersion

In [34]:
number_of_students = 150

# Statistics
stats_mean = 78
stats_sd = 8

# Algebra
algebra_mean = 73
algebra_sd = 7.6

if stats_sd > algebra_sd:
    print(f'The statistics subject has greater absolute dispersion, as indicated by its higher standard deviation ({stats_sd} vs. {algebra_sd}).')
elif stats_sd < algebra_sd:
    print(f'The algebra subject has greater absolute dispersion, as indicated by its higher standard deviation ({algebra_sd} vs. {stats_sd}).')
else:
    print(f'Neither of the subject has the greater absolute dispersion due to both having the same standard deviation.')

The statistics subject has greater absolute dispersion, as indicated by its higher standard deviation (8 vs. 7.6).


### 2b. In which subject was there the greater (b) relative dispersion?

In [36]:
# Statistics
stats_cv = round(stats_sd/stats_mean, 4)

# Algebra
algebra_cv = round(algebra_sd/algebra_mean, 4)

if stats_cv > algebra_cv:
    print(f'The statistics subject has greater relative dispersion, as indicated by its higher relative dispersion ({stats_cv} vs {algebra_cv}).')
elif stats_cv < algebra_cv:
    print(f'The algebra subject has greater relative dispersion, as indicated by its higher relative dispersion ({algebra_cv} vs {stats_cv}).')
else:
    print(f'Neither of the subject has the greater relative dispersion due to both having the same value of relative dispersion.')

The algebra subject has greater relative dispersion, as indicated by its higher relative dispersion (0.1041 vs 0.1026).


### 3. Prove that the mean and standard deviation of a set of standard scores are equal to 0 and 1, respectively. Use the following problem to illustrate this: Convert the set 6, 2, 8, 7, 5 into standard scores.

#### a. Calculate the Mean (𝜇) of the Original Data Set

In [37]:
data_set = [6, 2, 8, 7, 5]

In [39]:
data_set_mean = np.mean(data_set)
print(f'Data set mean: {data_set_mean}')

Data set mean: 5.6


$$
μ = \frac{6+2+8+7+5}{5} = \frac{28}{5} = 5.6
$$

#### b. Calculate the Standard Deviation (𝜎) of the Original Data Set

In [59]:
# Calculate Each Deviation from the Mean
dev_set = []
for i in range(len(data_set)):
    dev = round(data_set[i] - data_set_mean, 1)
    dev_set.append(dev)

print(f'Deviation: {dev_set}')

# Square Each Deviation
dev_squared = []
for i in range(len(dev_set)):
    square = round(dev_set[i] ** 2, 2)
    dev_squared.append(square)
    
print(f'Square Deviation: {dev_squared}')
    
# Calculate the Variance
var = sum(dev_squared)/len(dev_squared)

print(f'Variance: {var}')

# Calculate the Standard Deviation
sd = round(np.sqrt(var), 2)

print(f'Standard Deviation: {sd}')

Deviation: [0.4, -3.6, 2.4, 1.4, -0.6]
Square Deviation: [0.16, 12.96, 5.76, 1.96, 0.36]
Variance: 4.24
Standard Deviation: 2.06


##### Calculate Each Deviation from the Mean:
$$
6 - 5.6 = 0.4 \\
2 - 5.6 = -3.6 \\
8 - 5.6 = 2.4 \\
7 - 5.6 = 1.4 \\
5 - 5.6 = -0.6
$$

##### Square Each Deviation:
$$
0.4^2 = 0.16 \\
-3.6^2 = 12.96 \\
2.4^2 = 5.76 \\
1.4^2 = 1.96 \\
-0.6^2 = 0.36
$$

##### Calculate the Variance:
$$
Variance = \frac{0.16+12.96+5.76+1.96+0.36}{5} = \frac{21.2}{5} = 4.24
$$

##### Calculate the Standard Deviation:
$$
σ = \sqrt{4.24} ≈ 2.06
$$

#### c. Convert the Data Set into Standard Scores (z-scores)

In [70]:
z_scores = []
for i in range(len(data_set)):
    z_score = round((data_set[i] - data_set_mean)/sd, 3)
    z_scores.append(z_score)
    
print(f'Standard Scores (z-scores): {z_scores}')

Standard Scores (z-scores): [0.194, -1.748, 1.165, 0.68, -0.291]


Using the formula:
$$
z = \frac{(X - μ)}{σ}
$$

First Score:
$$
z = \frac{6−5.6}{2.06} ≈ \frac{0.4}{2.06} ≈ 0.194
$$

Second Score:
$$
z = \frac{2−5.6}{2.06} ≈ \frac{-3.6}{2.06} ≈ -1.748
$$

Third Score:
$$
z = \frac{8−5.6}{2.06} ≈ \frac{2.4}{2.06} ≈ 1.165
$$

Fourth Score:
$$
z = \frac{7−5.6}{2.06} ≈ \frac{1.4}{2.06} ≈ 0.68
$$

Fifth Score:
$$
z = \frac{5−5.6}{2.06} ≈ \frac{-0.6}{2.06} ≈ -0.291
$$

#### d. Calculate the Mean and Standard Deviation of the z-scores

In [84]:
# Calculate the Mean
z_scores_mean = round(sum(z_scores)/len(z_scores), 5)

print(f'Mean of Standard Scores (z-scores): {z_scores_mean}')

Mean of Standard Scores (z-scores): 0.0


##### Mean of Standard Scores (z-scores):
$$
Mean = \frac{0.194+(−1.749)+1.165+0.679+(−0.291)}{5} ≈ \frac{0}{5} ≈ 0
$$

##### Standard Deviation of z-scores:
Since we have calculated z-scores based on the standard deviation of the original data, the standard deviation of the z-scores is always:
$$
σ_z=1
$$

##### Conclusion:
The mean of the z-scores is 0. \
The standard deviation of the z-scores is 1.

### 3a. Three masses are measured as 20.48, 35.97, and 62.34 g, with standard deviations of 0.21, 0.46, and 0.54 g, respectively. Find the (a) mean and (b) standard deviation of the sum of the masses.

In [88]:
# Masses
m1 = 20.48
m2 = 35.97
m3 = 62.34

# Standard Deviations
sd1 = 0.21
sd2 = 0.46
sd3 = 0.54

# Mean
masses_mean = round((m1+m2+m3)/3, 2)
print(f'Mean of the Masses: {masses_mean} grams')

# standard deviation of the sum of the masses
m_sd1 = sd1 ** 2
m_sd2 = sd2 ** 2
m_sd3 = sd3 ** 2

var_sum = m_sd1 + m_sd2 + m_sd3

final_sd = round(np.sqrt(var_sum), 2)

print(f'Standard deviation of the sum of the masses: {final_sd} grams')

Mean of the Masses: 39.6 grams
Standard deviation of the sum of the masses: 0.74 grams


### 5. The credit hour distribution at Metropolitan Technological College is as follows

In [106]:
data_MTC = {
    'x': [6, 9, 12, 15, 18],
    'p(x)': [0.1, 0.2, 0.4, 0.2, 0.1]
}

df_MTC = pd.DataFrame(data_MTC)
print(df_MTC.to_string(index=False))

 x  p(x)
 6   0.1
 9   0.2
12   0.4
15   0.2
18   0.1


### Find 𝜇 and $𝜎^2$ . Give the 25 (with replacement) possible samples of size 2, their means, and their probabilities.


In [112]:
import itertools

# Given data
x_values = df_MTC['x']
p_values = df_MTC['p(x)']

# Calculate the population mean (mu)
mu = sum([x * p for x, p in zip(x_values, p_values)])

# Step 2: Calculate the population variance (sigma^2)
variance = sum([(x - mu) ** 2 * p for x, p in zip(x_values, p_values)])

# Generate all possible samples of size 2 (with replacement)
samples = list(itertools.product(x_values, repeat=2))

# Calculate the mean and probability for each sample
sample_means = [(x1 + x2) / 2 for x1, x2 in samples]
sample_probabilities = [
    p_values[df_MTC.index[df_MTC['x'] == x1][0]] * p_values[df_MTC.index[df_MTC['x'] == x2][0]]
    for x1, x2 in samples
]

# Combine results into a DataFrame
results_df = pd.DataFrame({
    'Sample': samples,
    'Mean': sample_means,
    'Probability': sample_probabilities
})

# Display the results
print(results_df)

      Sample  Mean  Probability
0     (6, 6)   6.0         0.01
1     (6, 9)   7.5         0.02
2    (6, 12)   9.0         0.04
3    (6, 15)  10.5         0.02
4    (6, 18)  12.0         0.01
5     (9, 6)   7.5         0.02
6     (9, 9)   9.0         0.04
7    (9, 12)  10.5         0.08
8    (9, 15)  12.0         0.04
9    (9, 18)  13.5         0.02
10   (12, 6)   9.0         0.04
11   (12, 9)  10.5         0.08
12  (12, 12)  12.0         0.16
13  (12, 15)  13.5         0.08
14  (12, 18)  15.0         0.04
15   (15, 6)  10.5         0.02
16   (15, 9)  12.0         0.04
17  (15, 12)  13.5         0.08
18  (15, 15)  15.0         0.04
19  (15, 18)  16.5         0.02
20   (18, 6)  12.0         0.01
21   (18, 9)  13.5         0.02
22  (18, 12)  15.0         0.04
23  (18, 15)  16.5         0.02
24  (18, 18)  18.0         0.01
